### PR TITLE
JSDoc support, Bundle mode fix, and more

### LINKED ISSUE #49: Bundle mode doesn't work
I ran into this while using `--mode bundle`, but saw the same issue is present in the examples.

If I apply the sql files from the [hello-bundle](https://github.com/divyenduz/plv8ify/tree/20edcd11c9b445582ca69de2ca79ad53233c80c3/examples/hello-bundle) and then run
```sql
SELECT myScopehello();
```

I get `ReferenceError: hello is not defined`.

It seems like there's a few issues happening here:
- functions declared in the `myScope_init` are not made available in the global scope.
  - [plv8 documentation is consistent with this](https://github.com/plv8/plv8/blob/83296584ba9cf3c1cedab37fe7cf0cb078264d4e/doc/plv8.md?plain=1#L263-L267)
  
    > The function will be internally defined such that:
    > 
    > (function(arg1, arg2, ..){
    >     $funcbody$
    > })

- functions are checking for `globalThis.myScope === undefined`, but the init function does nothing to define `globalThis.myScope`
- if you don't use the `--scope-prefix` option (and let it be the default of `(empty string)`, then the expression is malformed: `if (globalThis. === undefined) plv8.execute('SELECT myScope_init();');`

-----

Docker Image I'm using for PG - docker.io/library/postgres:14.4-alpine
plv8ify - 0.0.56
plv8 - 3.1.7

### PR DESCRIPTION
**Pull request by ferm10n**
Hey! I was planning to implement something like `plv8ify` for me and my team, and found that this project is already doing _mostly_ what we need. I know it's ideal to have separate PRs for separate fixes/features, but it was just easier to put everything into one PR for now. I'm happy to split them up if needed, just lmk.

Here's the gist:
- added support for JSDoc tags to decorate functions instead of magic comments
  - this is neat because it uses the jsdoc parsing already provided by ts-morph
  - I tried to keep the format consistent with similar JSDoc tags (`@param` has type first, then param name)
  - using `_` instead of `-` because syntax highlighting doesn't recognize `@plv8ify-whatever` as full tag
  - I think using JSDoc tags would be preferable for most developers, and I updated the examples and the README to reflect that. parsing comments still works for legacy reasons but it's not presented in the README
- type mapping can be controlled on a per-function basis
  - return: `/** @plv8ify_return {<SQL TYPE>} */` or `//@plv8ify-return <SQL TYPE>`
  - param: `/** @plv8ify_param {<SQL TYPE>} <PARAM NAME> */` or `//@plv8ify-param <PARAM NAME> <SQL TYPE>`
    - NOTE: order for the sql type and param name is flipped for jsdoc VS magic comment... maybe this isn't desirable but it might become a non-issue if you decide to remove comment support altogether?
- the examples with custom types weren't using their custom types, fixed
- fixed #49 via use of a symbol
  - I chose the symbol because it's pretty much guaranteed to be unique and avoid collisions
- addressed some `TODO` comments and fixed the "risky" type assertions
- consolidated all the logic for processing jsdoc / comments into a single function that returns a config object describing how the fn should be converted to SQL
- I didn't like how the `TSFunction.returnType` was modified to be a sql type, because it was ambiguous not knowing if the type had been mapped yet. So I decided to keep the value in `returnType` and introduce a separate `sqlReturnType`

### CODE REVIEW

**Review by divyenduz — APPROVED** [review]
Thanks, this looks fantastic

- JSDoc > magic comments, didn't know that ts-morph can parse them. This is a breaking change but since we are still 0.x.y, this is fine. Feel free to remove legacy comment parsing in a separate PR (I won't block you on that and merge this one) or I will do that later!
- "type mapping can be controlled on a per-function basis" - I like this

Thanks for fixing the examples and other fixes/improvements! Really appreciate it.

**Comment by divyenduz** [comment]
Looks like tests are failing: https://github.com/divyenduz/plv8ify/actions/runs/9492758988/job/26173844715?pr=50
This will block the automate release even if I merge this, looking into fixing them after the merge. Will post the next released version here once I am done.

**Comment by divyenduz** [comment]
These changes are live in [0.0.57](https://github.com/divyenduz/plv8ify/commit/d0513f1ab4e00a28e0ed90673bf1b4183ce83eb7) 🚀 . Thanks again for all the work!

**Comment by ferm10n** [comment]
Fantastic! Thanks for moving on this so quickly and getting a new release published 💙

**Pull request merged**
